packages/pools/src/concentrated/fetch-pool-amount-provider.ts

```diff
@@ -0,0 +1,71 @@
+import { Int } from "@keplr-wallet/unit";
+
+import { AmountsDataProvider, ConcentratedLiquidityPool } from "./pool";
+
+type BankAmountResponse = {
+  balances: {
+    amount: string;
+    denom: string;
+  }[];
+};
+
+/** Provider of bank balances from bank module by bech32Address. Expected by concentrated liquidity pool. */
+export class FetchPoolAmountProvider implements AmountsDataProvider {
+  /** Address => Response */
+  protected amountsCache = new Map<
+    string,
+    Awaited<ReturnType<AmountsDataProvider["getPoolAmounts"]>>
+  >();
+
+  constructor(
+    protected readonly baseNodeUrl: string,
+    protected readonly poolId: string,
+    protected readonly cacheDurationMs = 1000 * 30, // 30 seconds
+    protected readonly amountFetcher: (
+      bech32Address: string
+    ) => Promise<BankAmountResponse> = async (bech32Address) => {
+      if (fetch === undefined)
+        throw new Error("Fetch method must be available in the environment");
+
+      const res = await fetch(
```

Comment:
Elsewhere in our app, we are using axios. Is there a reason to use the native `fetch` here?